pict-lib/pict/private/utils.rkt

```diff
@@ -1394,3 +1484,435 @@
+                      #:nose-color [nose-color "pink"]
+                      #:nose [nose (equilateral-triangle (/ width 4) (/ height 4) #:angle pi #:color nose-color)]
+                      #:lip-color [lip-color (make-object color% 240 140 0)]
+                      #:lip-border-color [lip-border-color fur-border-color]
+                      #:lip-border-width [lip-border-width 1]
+                      #:happy? [happy? #f]
+                      #:eyes [eyes
+                              (if happy?
+                                  (happy-eyes (/ width 6) (/ height 6) #:color eye-color)
+                                  (filled-ellipse (/ width 6) (/ height 6) #:color eye-color))]
+                      #:left-eye [left-eye #f]
+                      #:right-eye [right-eye #f]
+                      #:whisker-length [whisker-length (* 3/8 width)]
+                      #:whisker-droop [whisker-droop (* 1/6 height)]
+                      #:whisker-color [whisker-color (make-object color% 100 40 50)]
+                      #:whisker-width [whisker-width 1]
+                      #:whisker-inset? [whisker-inset? #t]
+                      #:border-width [border-width 1])
+  (define-values (face coord-fn)
+    (cat-silhouette width height
+                    #:left-ear-extent left-ear-extent
+                    #:left-ear-arc left-ear-arc
+                    #:left-ear-angle left-ear-angle
+                    #:right-ear-extent right-ear-extent
+                    #:right-ear-arc right-ear-arc
+                    #:right-ear-angle right-ear-angle
+                    #:color fur-color
+                    #:border-color fur-border-color
+                    #:border-width border-width))
+
+  (match-define (cons pre-center-x center-y) (coord-fn (cons 0 0)))
+
+  (define left-whisker-center-offset
+    (* -3/4 (pict-width nose)))
+  (define left-whisker-extent-from-center
+    (+ (- whisker-length) left-whisker-center-offset))
+  (define left-whisker-inset-amount
+    (if whisker-inset?
+        (max 0 (- (car (coord-fn (cons left-whisker-extent-from-center 0)))))
+        0))
+
+  (define right-whisker-center-offset
+    (* 3/4 (pict-width nose)))
+  (define right-whisker-extent-from-center
+    (+ whisker-length right-whisker-center-offset))
+  (define right-whisker-inset-amount
+    (if whisker-inset?
+        (max 0 (- (car (coord-fn (cons right-whisker-extent-from-center 0)))
+                  (pict-width face)))
+        0))
+
+  ;; If we don't show the whiskers, we don't need to inset the face
+  ;; in order to draw them past the extent of the cat silhouette.
+  (define-values (center-x face*)
+    (cond [(not whisker-width) (values pre-center-x face)]
+          [else
+           (define inset-face
+             (inset face left-whisker-inset-amount 0
+                    right-whisker-inset-amount 0))
+           (values (+ pre-center-x left-whisker-inset-amount)
+                   inset-face)]))
+
+  (define face-and-nose
```

Comment:
Can the nose come after the lips?

(So, in case they overlap, it looks more like the normal real-life cat)

```
(standard-cat 60 60 #:nose (standard-fish 20 20))
```